src/phoenix/server/api/mutations/auth_mutations.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+from datetime import timedelta
+
+import strawberry
+from sqlalchemy import select
+from strawberry.types import Info
+
+from phoenix.auth import is_valid_password
+from phoenix.db import models
+from phoenix.server.api.context import Context
+from phoenix.server.api.mutations.auth import HasSecret
+
+PHOENIX_ACCESS_TOKEN_COOKIE_NAME = "phoenix-access-token"
+PHOENIX_ACCESS_TOKEN_COOKIE_MAX_AGE_IN_SECONDS = int(timedelta(days=31).total_seconds())
+
+
+@strawberry.input
+class LoginMutationInput:
+    email: str
+    password: str
+
+
+@strawberry.type
+class AuthMutationPayload:
+    success: bool
+
+
+@strawberry.type
+class AuthMutationMixin:
+    @strawberry.mutation(permission_classes=[HasSecret])  # type: ignore
+    async def login(
+        self,
+        info: Info[Context, None],
+        input: LoginMutationInput,
+    ) -> AuthMutationPayload:
+        async with info.context.db() as session:
+            if (
+                user := await session.scalar(
+                    select(models.User).where(models.User.email == input.email)
+                )
+            ) is None or (password_hash := user.password_hash) is None:
+                return AuthMutationPayload(success=False)
```

Comment:
This should throw as an error - we can figure out the nuances of the error extension but it should not succeed